ldap_auth_provider.py

```diff
@@ -628,14 +651,13 @@ async def _map_login_to_upn(self, username):
                 domain += "." + ldap_root_domain
         elif "/" in username:
             (login, domain) = username.lower().rsplit("/", 1)
-        else:
-            if not self.ldap_default_domain:
-                logger.info(
-                    'No LDAP separator "/" was found in uid "%s" '
-                    "and LDAP default domain was not configured.",
-                    username,
-                )
-                raise ActiveDirectoryUPNException()
+        elif domain is None:
```

Comment:
I think this would be clearer as `elif not self.ldap_default_domain`. Would mypy accept that?

(Otherwise it feels like a bug since `domain` is set in the above if-clauses, although I don't think it is a bug. This code could probably use comments...)

Reply:
I'm fairly suspicious of this code as well, but I'm not in-the-know.

I thought Windows names looked like `domain\user`; now I notice it seems like `user/domain` might be acceptable as well.
However, we only do this code block for backslash-style logins:
```python
            ldap_root_domain = await self._fetch_root_domain()
            if ldap_root_domain and not domain.endswith(ldap_root_domain):
                domain += "." + ldap_root_domain
```
That sounds somewhat suspicious, but again, I'm really not in the know about Active Directory (and barely about Windows). As you say, comments are in demand in this region of the code :-).

However your suggestion seems to make sense and I can't see anything wrong with it, so let's go with that

Reply:
Ah, doing it your way requires `assert domain is not None` after the block. But that's how it is — worry not